contracts/token/ERC20/extensions/ERC20Permit.sol

```diff
@@ -24,6 +24,16 @@ abstract contract ERC20Permit is ERC20, IERC20Permit, EIP712, Nonces {
     bytes32 private constant _PERMIT_TYPEHASH =
         keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
 
+    /**
+     * @dev Permit deadline has expired.
+     */
+    error ERC2612ExpiredDeadline(uint256 deadline);
```

Comment:
I would rename this to `ERC2612ExpiredSignature(uint256 deadline)` for consistency with `Votes`.